nav2_amcl/src/amcl_node.cpp

```diff
@@ -133,7 +133,7 @@ AmclNode::AmclNode()
     };
   set_map_srv_ = create_service<nav_msgs::srv::SetMap>("set_map", handle_set_map_callback);
 
-
+  custom_qos_profile = rmw_qos_profile_sensor_data;
```

Comment:
We should probably be using the new style for specifying QoS settings. 

Here's an example

https://github.com/ros-planning/navigation2/blob/6d837b17cdd81eaddca42ffde5946fbcad9f2686/nav2_robot/src/robot.cpp#L109

However, I see you are using it for message filters and am not sure it's supported there.

Reply:
I tried this and the message filter interface doesn't support the `rclcpp::SensorDataQoS()` method like you feared.